src/main/java/seedu/financialplanner/commands/BudgetCommand.java

```diff
@@ -53,38 +51,98 @@ public BudgetCommand(RawCommand rawCommand) throws FinancialPlannerException {
+        }
+
+        if (!rawCommand.extraArgs.containsKey("b")) {
+            logger.log(Level.WARNING, "Missing argument /b in command");
+            throw new IllegalArgumentException("Missing /b argument.");
         }
     }
 
     @Override
     public void execute() {
-        assert command.equals("set") || command.equals("update") : "Command should be set or update only";
+        assert command.equals("set") || command.equals("update") || command.equals("delete") ||
+                command.equals("reset") || command.equals("view");
 
-        Ui ui = Ui.getInstance();
         switch (command) {
         case "set":
-            logger.log(Level.INFO, "Setting budget");
-            Budget.setBudget(budget);
-            ui.showMessage("A monthly budget of " + Budget.getInitialBudgetString()
-                    + " has been set.");
+            setBudget();
             break;
         case "update":
-            logger.log(Level.INFO, "Updating budget");
-            ui.printBudgetBeforeUpdate();
-            Budget.updateBudget(budget);
-            ui.printBudgetAfterUpdate();
+            updateBudget();
+            break;
+        case "delete":
+            deleteBudget();
+            break;
+        case "reset":
+            resetBudget();
+            break;
+        case "view":
+            viewBudget();
             break;
         default:
             logger.log(Level.SEVERE, "Unreachable default case reached");
             ui.showMessage("Unknown command.");
         }
     }
+
+    private void viewBudget() {
+        if (Budget.hasBudget()) {
+            ui.printBudget();
+        } else {
+            ui.showMessage("There is no existing budget.");
+        }
+    }
+
+    private void resetBudget() {
+        if (Budget.getInitialBudget() != Budget.getCurrentBudget()) {
+            if (Budget.getInitialBudget() > Cashflow.getBalance()) {
+                Budget.setInitialBudget(Cashflow.getBalance());
+                ui.showMessage("Since initial budget exceeds current balance, " +
+                        "budget will be reset to current balance.");
+            }
+            Budget.resetBudget();
+            ui.printResetBudget();
+        } else {
+            ui.showMessage("Budget has not been spent yet.");
+        }
+    }
+
+    private void deleteBudget() {
+        if (Budget.hasBudget()) {
+            Budget.deleteBudget();
+            ui.printDeleteBudget();
+        } else {
+            ui.showMessage("Budget has not been set yet.");
+        }
+    }
+
+    private void updateBudget() {
+        logger.log(Level.INFO, "Updating budget");
+        ui.printBudgetBeforeUpdate();
+        Budget.updateBudget(budget);
+        ui.printBudgetAfterUpdate();
+    }
+
+    private void setBudget() {
+        logger.log(Level.INFO, "Setting budget");
+        Budget.setBudget(budget);
+        ui.showMessage("A monthly budget of " + Budget.getInitialBudgetString()
```

Comment:
maybe extract this method further into showMonthlyBudget in UI?

Reply:
ok will note for the future